files/image_config/process-reboot-cause/process-reboot-cause

```diff
@@ -73,40 +93,55 @@ def main():
     try:
         import sonic_platform
 
-        # Check if the previous reboot was caused by hardware
-        platform = sonic_platform.platform.Platform()
-
-        chassis = platform.get_chassis()
-
-        hardware_reboot_cause, optional_details = chassis.get_reboot_cause()
+        proc_cmdline_reboot_cause = None
 
-        if hardware_reboot_cause == chassis.REBOOT_CAUSE_NON_HARDWARE:
-            # The reboot was not caused by hardware. If there is a REBOOT_CAUSE_FILE, it will
-            # contain any software-related reboot info. We will use it as the previous cause.
+        # 1. Check if the previous reboot was warm/fast reboot by testing whether there is "fast|fastfast|warm" in /proc/cmdline
+        #    If yes, the content of /hosts/reboot-cause/reboot-cause.txt will be treated as the reboot cause
+        if is_warmfast_reboot_from_proc_cmdline():
             if os.path.isfile(REBOOT_CAUSE_FILE):
-                cause_file = open(REBOOT_CAUSE_FILE, "r")
-                previous_reboot_cause = cause_file.readline().rstrip('\n')
-                cause_file.close()
-            # If it is FirstTime Boot and previous_reboot_cause is unknown
-            # and hardware_reboot cause is non_hardware then
-            # Update the reboot cause as required
-            if os.path.isfile(FIRST_BOOT_PLATFORM_FILE):
-                if (previous_reboot_cause == UNKNOWN_REBOOT_CAUSE):
-                     previous_reboot_cause = UNKNOWN_REBOOT_CAUSE
-                os.remove(FIRST_BOOT_PLATFORM_FILE)
-        elif hardware_reboot_cause == chassis.REBOOT_CAUSE_HARDWARE_OTHER:
-            previous_reboot_cause = "{} ({})".format(hardware_reboot_cause, optional_details)
+                with open(REBOOT_CAUSE_FILE, "r") as cause_file:
+                    proc_cmdline_reboot_cause = cause_file.readline().rstrip('\n')
+            else:
+                # /proc/cmdline says it's a warm/fast reboot but /host/reboot-cause.txt doesn't exist.
+                # report an error.
+                log_error("/proc/cmdline indicates a fast/warm reboot but {} doesn't exist".format(REBOOT_CAUSE_DIR))
```

Comment:
other than report error,

I think we should still assign the 

proc_cmdline_reboot_cause = warm or fast

we should treat proc cmdline as truth.

Reply:
Done